patches/api/0441-Expose-native-map-methods-in-PDC.patch

```diff
@@ -0,0 +1,288 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Xander <xander.dekeijzer@student.hu.nl>
+Co-authored-by: TheRealRyGuy <ryan@ryguy.me>
+Date: Tue, 3 Oct 2023 07:28:50 +0200
+Subject: [PATCH] Expose native map methods in PDC
+
+
+diff --git a/src/main/java/org/bukkit/persistence/PersistentDataContainer.java b/src/main/java/org/bukkit/persistence/PersistentDataContainer.java
+index 57609b7793122e135fa0c3b926500849379637b2..b221b97fd540b1f16aea3d6fce72d8a09c7a5885 100644
+--- a/src/main/java/org/bukkit/persistence/PersistentDataContainer.java
++++ b/src/main/java/org/bukkit/persistence/PersistentDataContainer.java
+@@ -9,7 +9,7 @@ import org.jetbrains.annotations.Nullable;
+  * This interface represents a map like object, capable of storing custom tags
+  * in it.
+  */
+-public interface PersistentDataContainer {
++public interface PersistentDataContainer extends Iterable<net.kyori.adventure.key.Key> {
```

Comment:
I don't think its appropriate to implement Iterable, especially when the iterator's remove doesn't actually remove the key from the PDC because the `keySet` set isn't backed by the pdc map.